### ISSUE TITLE
MD055 and MD057 report false positives for tables inside blockquotes

### ISSUE THREAD

**Issue by ssokolow** [markdownlint/markdownlint#509]
### Environment
```
% lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.5 LTS
Release:        22.04
Codename:       jammy
```

**MDL Version**
```
% mdl --version
0.13.0
```

### Expected Behavior
It shouldn't say anything.

### Actual Behavior
```
test.md:1046: MD055 Table row doesn't begin/end with pipes
test.md:1047: MD055 Table row doesn't begin/end with pipes
test.md:1048: MD055 Table row doesn't begin/end with pipes
test.md:1049: MD055 Table row doesn't begin/end with pipes
test.md:1047: MD057 Table has missing or invalid header separation (second row)
```

## Replication Case

```
> | foo | bar  |
> | --- | ---- |
> | baz | quux |
```

Here's the snippet that triggers it in production:

```
## Emoji {#sec-emoji}

This renderer supports all
[GitHub emoji shortcodes](https://github.com/ikatyang/emoji-cheat-sheet/blob/master/README.md)
supported by the [`emoji`](https://lib.rs/crates/emojis) crate.

Emoji are not rendered inside code spans (`:smile:`) or code blocks.

> [!WARNING]
>
> GitHub has rudely chosen to use the convenient `:tm:` for a forced-emoji
> variation of the trademark symbol. As such, it has been reassigned to
> `:tmoji:` to free up the more useful name:
>
> | Shortcode | Result         |     | Shortcode     | Result              |
> | --------- | -------------- | --- | ------------- | ------------------- |
> | `:(c):`   | :(c): (normal) |     | `:copyright:` | :copyright: (emoji) |
> | `:tm:`    | :tm: (normal)  |     | `:tmoji:`     | :tmoji: (emoji)     |
```

![Screenshot 2024-09-30 at 20-04-31 Syntax Reference](https://github.com/user-attachments/assets/223b3342-819c-4418-ae11-5562e6152a26)

The parser I'm referencing against when I say it renders fine is `pulldown-cmark` v0.12.1 with GFM support turned on and this iterator adaptor on it to allow that blank line after `[!WARNING]` so that Prettier won't trash the admonition syntax:

```rust
/// An iterator adapter to collapse together the incorrectly split BlockQuote elements produced by
/// pulldown-cmark when using blockquote type tags/admonitions.
pub struct FixPulldownCmarkIssue890<'a, T: Iterator> {
    inner: std::iter::Peekable<T>,
    buffer: Vec<Option<Event<'a>>>,
}
impl<'a, T> FixPulldownCmarkIssue890<'a, T>
where
    T: Iterator<Item = Event<'a>>,
{
    pub fn new(iterator: T) -> Self {
        Self { inner: iterator.peekable(), buffer: Vec::new() }
    }
}

impl<'a, T> Iterator for FixPulldownCmarkIssue890<'a, T>
where
    T: Iterator<Item = Event<'a>>,
{
    type Item = Event<'a>;

    fn next(&mut self) -> Option<Self::Item> {
        loop {
            match (self.inner.peek(), self.buffer.as_slice()) {
                // If we see 3 after accumulating 1&2, drop 2&3 and return 1.
                (
                    Some(Event::Start(Tag::BlockQuote(None))),
                    [Some(Event::Start(Tag::BlockQuote(Some(_)))), Some(Event::End(TagEnd::BlockQuote(_)))],
                ) => {
                    let _ = self.inner.next();
                    let e = self.buffer.swap_remove(0);
                    self.buffer.clear();
                    return e;
                },
                // If we see 2 and we've accumulated 1, buffer it and go around again
                (
                    Some(Event::End(TagEnd::BlockQuote(_))),
                    [Some(Event::Start(Tag::BlockQuote(Some(_))))],
                ) => {
                    self.buffer.push(self.inner.next());
                },
                // If we see 1 and the buffer is empty, buffer it and go around again
                (Some(Event::Start(Tag::BlockQuote(Some(_)))), []) => {
                    self.buffer.push(self.inner.next());
                },
                // Otherwise, if the buffer is empty, just pass it through
                (_, []) => return self.inner.next(),
                // Otherwise, drain the buffer
                (_, [_, ..]) => return self.buffer.remove(0),
            }
        }
    }
}
```

(Prettier-on-save forcibly re-wraps the "first line" of the blockquote, so that blank line needs to be there to reconcile GFM admonition syntax with Prettier's opinionated formatting.)
